ghost/staff-service/lib/emails.js

```diff
@@ -0,0 +1,291 @@
+            return '';
+        }
+        const date = new Date(stripeDate * 1000);
+
+        return this.getFormattedDate(date);
+    }
+
+    /** @private */
+    getOfferData(offer) {
+        if (offer) {
+            let offAmount = '';
+            let offDuration = '';
+
+            if (offer.duration === 'once') {
+                offDuration = ', first payment';
+            } else if (offer.duration === 'repeating') {
+                offDuration = `, first ${offer.duration_in_months} months`;
+            } else if (offer.duration === 'forever') {
+                offDuration = `, forever`;
+            } else if (offer.duration === 'trial') {
+                offDuration = '';
+            }
+            if (offer.type === 'percent') {
+                offAmount = `${offer.amount}% off`;
+            } else if (offer.type === 'fixed') {
+                offAmount = `${this.getFormattedAmount({currency: offer.currency, amount: offer.amount})} off`;
+            } else if (offer.type === 'trial') {
+                offAmount = `${offer.amount} days free`;
+            }
+
+            return {
+                name: offer.name,
+                details: `${offAmount}${offDuration}`
+            };
+        }
+    }
+
+    get siteDomain() {
+        const [, siteDomain] = this.urlUtils.getSiteUrl()
+            .match(new RegExp('^https?://([^/:?#]+)(?:[/:?#]|$)', 'i'));
+
+        return siteDomain;
+    }
+
+    get membersAddress() {
+        // TODO: get from address of default newsletter?
+        return `noreply@${this.siteDomain}`;
+    }
+
+    get fromEmailAddress() {
+        return `ghost@${this.siteDomain}`;
+    }
+
+    // TODO: duplicated from services/members/config - exrtact to settings?
+    get supportAddress() {
+        const supportAddress = this.settingsCache.get('members_support_address') || 'noreply';
+
+        // Any fromAddress without domain uses site domain, like default setting `noreply`
+        if (supportAddress.indexOf('@') < 0) {
+            return `${supportAddress}@${this.siteDomain}`;
+        }
+
+        return supportAddress;
+    }
+
+    get notificationFromAddress() {
+        return this.supportAddress || this.membersAddress;
+    }
+
+    extractInitials(name = '') {
+        const names = name.split(' ');
+        const initials = names.length > 1 ? [names[0][0], names[names.length - 1][0]] : [names[0][0]];
+        return initials.join('').toUpperCase();
+    }
+
+    async sendMail(message) {
+        if (process.env.NODE_ENV !== 'production') {
+            this.logging.warn(message.text);
+        }
+
+        let msg = Object.assign({
+            from: this.fromEmailAddress,
+            forceTextContent: true
+        }, message);
+
+        return this.mailer.send(msg);
+    }
+
+    async renderEmailTemplate(templateName, data) {
+        const htmlTemplateSource = await fs.readFile(path.join(__dirname, './email-templates/', `${templateName}.hbs`), 'utf8');
```

Comment:
Can we cache this file so we don't have to read from disk each time?

Reply:
Probably yea, will have a quick look at this later today or write it up so we don't miss it.